docs/TextRegion.md

```diff
@@ -0,0 +1,86 @@
+
+![](tcols-paragraphs.png)
+
+The graphic shows shows the relationship of page, text areas and paragraphs (with varying alignment) for the example of a two-column layout.
+
+
+### Text Start Position ###
+
+When rendering, the vertical start position of the text will be at the lowest one out of the current y position, the top of the region (if it has a defined top), or the top margin of the page. The horizontal start position will be either at the current x position if that lies within the boundaries of the region/column, or at the left edge of the region. In both horizontal and vertical positioning, regions with multiple columns may follow additional rules and restrictions.
+
+
+### Interaction between Regions ###
+
+Several region instances can exist at the same time. But only one of them can act as context manager at any given time. It is not currently possible to activate them recursively. But it is possible to use them intermittingly. This will probably most often make sense between a columnar region and a table or a graphic. You may have some running text ending at a given height, then insert a table/graphic, and finally continue the running text at the new height below the table within the existing column(s).
+
+
+### Common parameters ###
+
+All types of text regions have the following constructor paramters in common:
+
+* text (str, optional) - text content to add to the region. This is a convenience parameter for cases when all text is available in one piece, and no partition into paragraphs (possibly with different parameters) is required. (Default: None)
+* align (Align/str, optional) - the horizontal alignment of the text in the region. (Default: Align.L)
+* line_height (float, optional) - This is a factor by which the line spacing will be different from the font height. It works similar to the attribute of the same name in HTML/CSS. (default: 1.0)
+* print_sh (bool, optional) - Treat a soft-hyphen (\\u00ad) as a printable character, instead of a line breaking opportunity. (Default: False)
+* skip_leading_spaces (default: False) - This flag is primarily used by `write_html()`, but may also have other uses. It removes all space characters at the beginning of each line.
+* wrapmode (default "WORD") - 
+
+All of those values can be overriden for each individual paragraph.
+
+
+### Common methods ###
+
+* paragraph() [see characteristics parameters below] - establish a new paragraph in the text. The text added to this paragraph will start on a new line.
+* write(text: str, link: = None) - write text to the region. This is only permitted when no explicit paragraph is currently active.
+* ln(h: float = None) - Start a new line moving either by the current font height or by the parameter "h". Only permitted when no explicit paragraph is currently active.
+* render() - if the region is not used as a context manager with "with", this method must be called to actually process the added text.
```

Comment:
```suggestion
* `.paragraph()` [see characteristics parameters below] - establish a new paragraph in the text. The text added to this paragraph will start on a new line.
* `.write(text: str, link: = None)` - write text to the region. This is only permitted when no explicit paragraph is currently active.
* `.ln(h: float = None)` - Start a new line moving either by the current font height or by the parameter "h". Only permitted when no explicit paragraph is currently active.
* `.render()` - if the region is not used as a context manager with "with", this method must be called to actually process the added text.
```
About `.render()`, just a question our of curiosity: what would happen if this method is called a second time?

Reply:
`render()` consumes the rendered text, so nothing will happen, at least with columns (that have essentially infinite length and will consume everything).
With shaped regions, if there was more text than fits its outline, the remaining text will still around and can be passed on to another, new shape.

Reply:
Alright!